addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaReceiverHandlerFactory.java

```diff
@@ -59,7 +64,7 @@ protected ThingHandler createHandler(Thing thing) {
 
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        if (thingTypeUID.equals(YamahaReceiverBindingConstants.BRIDGE_THING_TYPE)) {
+        if (thingTypeUID.equals(YamahaReceiverBindingConstants.BRIDGE_THING_TYPE) && thing instanceof Bridge) {
```

Comment:
Before adding this check, we should figure out what is going on. I think we agree that such a check should not be necessary here (and if, that we would need some "else" branch to handle the faulty situation).